src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -135,25 +250,44 @@ CHIP_ERROR TestDACRevocationDelegateImpl::GetSerialNumberHexStr(const ByteSpan &
     return BytesToHexStr(serialNumber, outSerialNumberHexStr);
 }
 
-CHIP_ERROR TestDACRevocationDelegateImpl::GetIssuerNameBase64Str(const ByteSpan & certDer,
-                                                                 MutableCharSpan & outIssuerNameBase64String)
+CHIP_ERROR TestDACRevocationDelegateImpl::GetRDNBase64Str(const ByteSpan & certDer, MutableCharSpan & outRDNBase64String,
+                                                          bool isIssuer)
 {
-    uint8_t issuerBuf[kMaxCertificateDistinguishedNameLength] = { 0 };
-    MutableByteSpan issuer(issuerBuf);
+    uint8_t rdnBuf[kMaxCertificateDistinguishedNameLength] = { 0 };
+    MutableByteSpan rdn(rdnBuf);
 
-    ReturnErrorOnFailure(ExtractIssuerFromX509Cert(certDer, issuer));
-    VerifyOrReturnError(outIssuerNameBase64String.size() >= BASE64_ENCODED_LEN(issuer.size()), CHIP_ERROR_BUFFER_TOO_SMALL);
+    if (isIssuer)
+    {
+        ReturnErrorOnFailure(ExtractIssuerFromX509Cert(certDer, rdn));
+    }
+    else
+    {
+        ReturnErrorOnFailure(ExtractSubjectFromX509Cert(certDer, rdn));
+    }
 
-    uint16_t encodedLen = Base64Encode(issuer.data(), static_cast<uint16_t>(issuer.size()), outIssuerNameBase64String.data());
-    outIssuerNameBase64String.reduce_size(encodedLen);
+    VerifyOrReturnError(outRDNBase64String.size() >= BASE64_ENCODED_LEN(rdn.size()), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint16_t encodedLen = Base64Encode(rdn.data(), static_cast<uint16_t>(rdn.size()), outRDNBase64String.data());
+    outRDNBase64String.reduce_size(encodedLen);
     return CHIP_NO_ERROR;
 }
 
-bool TestDACRevocationDelegateImpl::IsCertificateRevoked(const ByteSpan & certDer)
+CHIP_ERROR TestDACRevocationDelegateImpl::GetIssuerNameBase64Str(const ByteSpan & certDer,
+                                                                 MutableCharSpan & outIssuerNameBase64String)
 {
-    static constexpr uint32_t maxIssuerBase64Len = BASE64_ENCODED_LEN(kMaxCertificateDistinguishedNameLength);
+    return GetRDNBase64Str(certDer, outIssuerNameBase64String, true /* isIssuer */);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetSubjectNameBase64Str(const ByteSpan & certDer,
+                                                                  MutableCharSpan & outSubjectNameBase64String)
+{
+    return GetRDNBase64Str(certDer, outSubjectNameBase64String, false /* isIssuer */);
+}
 
-    char issuerNameBuffer[maxIssuerBase64Len]                            = { 0 };
+// @param certDer Certificate being tested for revocation in
```

Comment:
"in" what?  It looks like this comment got truncated?

Reply:
"DER format" :-). Yeah, it was truncated.